cmd/flakechecker/flake_config.go

```diff
@@ -0,0 +1,103 @@
+package main
+
+import (
+	"fmt"
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v3"
+	"io"
+	"os"
+	"regexp"
+)
+
+// flakeDetectionPolicyConfig represents configuration used by flakechecker to evaluate failed tests.
+//
+// It contains the following fields:
+// jobNameRegex - regular expression for the name of the CI job that should be evaluated by flakechecker. (i.e. CI jobs for PRs should be evaluated, but not CI jobs for commits already merged to "main" branch)
+// className - class name of the test that should be isolated. Usually class name for Groovy tests, package name for golang tests, etc.
+// testNameRegex - regex used to match test names. Some test names contain detailed information (i.e. version 4.4.4), but we want to use ratio for all tests in that group (i.e. 4.4.z). Using a regex allow us to group tests as needed.
+// ratioJobName - CI job name that should be used for ratio calculation. i.e. we take CI runs for commits on "main" branch as input for evaluation of flake ratio.
+// ratioThreshold - maximum failure percentage that is used to distinguish a flaky test from a completely broken test. This information is usually fetched from historical executions and data collected in DB. If measured flakiness exceeds this threshold, we no longer want to suppress test failure, because we suspect it might have regressed above what we consider acceptable.
```

Comment:
Maybe these would fit better as comments for each individual field rather than here?